### ISSUE TITLE
Suppressed Babel plugin parsing error results in no linting

### ISSUE THREAD

**Issue by astone123** [stylelint/postcss-css-in-js#247]
### What steps are needed to reproduce the bug?

Run `stylelint "src/**/*.tsx"` 

### Relevant dependencies
```json
        "@emotion/styled": "11.6.0",
        "@stylelint/postcss-css-in-js": "0.37.2",
        "stylelint": "14.0.0",
        "stylelint-config-standard": "24.0.0",
        "stylelint-declaration-strict-value": "1.8.0",
        "stylelint-formatter-pretty": "2.1.1"
```

`babel.config.js` for `development`
```javascript
{
    development: {
      presets: [
        '@babel/preset-env',
        '@babel/preset-typescript',
        '@babel/preset-react'
      ],
      plugins: [
        [
          '@emotion',
          {
            sourceMap: true,
            cssPropOptimization: false
          }
        ],
        'babel-plugin-macros',
        '@babel/plugin-proposal-object-rest-spread',
        '@babel/plugin-proposal-class-properties',
        '@babel/plugin-proposal-private-methods'
      ]
    }
}
```

`.stylelintrc.json`
```json
{
    "defaultSeverity": "warning",
    "extends": "stylelint-config-standard",
    "overrides": [
        {
            "customSyntax": "@stylelint/postcss-css-in-js",
            "files": ["src/**/*.tsx"]
        },
        {
            "customSyntax": "postcss-scss",
            "files": ["sass/*/*.scss"]
        }
    ],
    "plugins": ["stylelint-declaration-strict-value"],
    "rules": {
        "property-no-unknown": true,
        "scale-unlimited/declaration-strict-value": [
            ["/color/", "/font/", "background"],
            {
                "severity": "warning"
            }
        ]
    }
}
```

### What did you expect to happen?

I expect there to be warnings in the output of Stylelint for my css-in-js styles using Emotion.

### What actually happened?

There are no warnings found by Stylelint, and the linting finishes successfully. There are errors during parsing that are suppressed in the following code block:

https://github.com/stylelint/postcss-css-in-js/blob/09af2eb5a1ba41279e099d095c268e170f02e88b/extract.js#L160-L165

If I uncomment the `console.error` line, I see the following error in the babel parser for each source
```
TypeError: p is not iterable
    at <projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:10658:43
    at Array.some (<anonymous>)
    at hasPlugin (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:10654:18)
    at validatePlugins (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:10694:9)
    at getParser (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:15287:5)
    at parse (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/parser/lib/index.js:15232:22)
    at parser (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/core/lib/parser/index.js:52:34)
    at parser.next (<anonymous>)
    at Function.parse (<projectRoot>/node_modules/@stylelint/postcss-css-in-js/node_modules/@babel/core/lib/parse.js:31:37)
    at parse.next (<anonymous>)
```

This happened when I was upgrading from Stylelint 13 -> 14. When using v13, I was using the bundled css-in-js parser, but in v14 that was removed, so now I'm trying to use `@stylelint/postcss-css-in-js` as a custom syntax for my `.tsx` files that use Emotion for css-in-js. But I don't see the style warnings that I was seeing before because of this parsing error with Babel.

It looks like the issue is that one or more plugins are not valid in this function in the Babel parser
```javascript
function hasPlugin(plugins, expectedConfig) {
  const [expectedName, expectedOptions] = typeof expectedConfig === "string" ? [expectedConfig, {}] : expectedConfig;
  const expectedKeys = Object.keys(expectedOptions);
  const expectedOptionsIsEmpty = expectedKeys.length === 0;
  return plugins.some(p => {
    if (typeof p === "string") {
      return expectedOptionsIsEmpty && p === expectedName;
    } else {
      const [pluginName, pluginOptions] = p;

      if (pluginName !== expectedName) {
        return false;
      }

      for (const key of expectedKeys) {
        if (pluginOptions[key] !== expectedOptions[key]) {
          return false;
        }
      }

      return true;
    }
  });
}
```

but I'm not sure why that would be.

**Comment by astone123** [comment]
If I log each plugin as Babel iterates through them, I can see where the error is being thrown. Whichever plugin I have listed **first** in my configuration is in a different format. From looking at the code, it looks like Babel expects the plugin to either be a string, or an array with the first element being the name, and the second being an options object. However, as you can see in the output, the Babel Emotion plugin is an object, which is neither a string, nor an array. The error is being thrown because Babel tries to deconstruct the object as if it were an array. 

Still trying to figure out why the plugins that I have defined are being loaded in as objects rather than a format that Babel expects.

```
PLUGIN typescript
PLUGIN objectRestSpread
PLUGIN [ 'decorators', { decoratorsBeforeExport: false } ]
PLUGIN typescript
PLUGIN objectRestSpread
PLUGIN [ 'decorators', { decoratorsBeforeExport: false } ]
PLUGIN classProperties
PLUGIN exportExtensions
PLUGIN asyncGenerators
PLUGIN functionBind
PLUGIN functionSent
PLUGIN dynamicImport
PLUGIN optionalCatchBinding
PLUGIN Plugin {
  key: '@emotion',
  manipulateOptions: [Function: manipulateOptions],
  post: undefined,
  pre: undefined,
  visitor: {
    _exploded: {},
    _verified: {},
    ImportDeclaration: { enter: [Array] },
    Program: { enter: [Array] },
    JSXAttribute: { enter: [Array] },
    CallExpression: { exit: [Array] }
  },
  parserOverride: undefined,
  generatorOverride: undefined,
  options: { sourceMap: true, cssPropOptimization: false }
}

```

**Comment by puku0x** [comment]
`hasPlugin` has been changed since https://github.com/babel/babel/pull/13982/commits/af9aaa170a116598691fbce8c1a084c1432d3fb8

**Comment by astone123** [comment]
Thanks @puku0x, I see that now... my project is using 
```
"@babel/core": "7.16.0"
```
which should be before that change, but in the stack trace, I can see that the version of `@babel/parser` that is being used is the one that is a dependency of `@stylelint/postcss-css-in-js`, which from looking at the package-lock file, looks like `v7.15.5`

https://github.com/stylelint/postcss-css-in-js/blob/master/package-lock.json#L307-L317

The change to `hasPlugin` shouldn't have been made yet for the versions that I'm using, so I'm still a little confused 🤔

**Comment by astone123** [comment]
I ended up adding `"@babel/core": "7.16.0"` to `resolutions` in my `package.json` file and this fixed the issue. Thanks for your help @puku0x

**Comment by mskelton** [comment]
Opened an issue with Babel here: https://github.com/babel/babel/issues/14082.

**Comment by mskelton** [comment]
These appear to be the problematic lines: https://github.com/stylelint/postcss-css-in-js/blob/master/extract.js#L145-L151

Not actually an issue with Babel, but this plugin passing bad options to Babel.  I removed those lines in a fork of this plugin and it worked no problem.  Not sure though why those are there in the first place, so not sure the solution for others.

**Comment by krzy789** [comment]
I'm also experiencing the same issue, any solutions other than forking?

**Comment by fdev** [comment]
> These appear to be the problematic lines: https://github.com/stylelint/postcss-css-in-js/blob/master/extract.js#L145-L151
> 
> Not actually an issue with Babel, but this plugin passing bad options to Babel. I removed those lines in a fork of this plugin and it worked no problem. Not sure though why those are there in the first place, so not sure the solution for others.

You're a lifesaver! :pray:

> I'm also experiencing the same issue, any solutions other than forking?

I'm using [patch-package](https://www.npmjs.com/package/patch-package) in combination with the following patch file:

```diff
diff --git a/node_modules/@stylelint/postcss-css-in-js/extract.js b/node_modules/@stylelint/postcss-css-in-js/extract.js
index 042580e..9e24e7f 100644
--- a/node_modules/@stylelint/postcss-css-in-js/extract.js
+++ b/node_modules/@stylelint/postcss-css-in-js/extract.js
@@ -87,7 +87,6 @@ const plugins = [
 	'jsx',
 	'typescript',
 	'objectRestSpread',
-	['decorators', { decoratorsBeforeExport: false }],
 	'classProperties',
 	'exportExtensions',
 	'asyncGenerators',
@@ -134,14 +133,6 @@ function loadBabelOpts(opts) {
 		}
 
 		opts[key] = fileOpts[key];
-
-		if (Array.isArray(fileOpts[key]) && Array.isArray(opts.parserOpts[key])) {
-			// combine arrays for plugins
-			opts.parserOpts[key] = opts.parserOpts[key].concat(fileOpts[key]);
-		} else {
-			// because some options need to be passed to parser also
-			opts.parserOpts[key] = fileOpts[key];
-		}
 	}
 
 	return opts;
```

Also, https://github.com/stylelint/postcss-css-in-js/issues/250 sounds very related.

**Comment by phyllisstein** [comment]
Thank you so much, @mskelton and @fdev. I did my complaining in #250 and missed this, but your patch did the trick for me.

**Comment by hudochenkov** [comment]
Should be fixed in just released `0.37.3`.
